Dockerfile

```diff
@@ -1,14 +1,44 @@
-FROM ruby:3.0.6
-WORKDIR /usr/src/app
-EXPOSE 3200
+# syntax=docker/dockerfile:1
 
-ENV OPENSSL_CONF /etc/ssl/
+ARG RUBY_VERSION=3.0.6
+FROM registry.docker.com/library/ruby:$RUBY_VERSION-slim as base
+
+WORKDIR /rails
+
+RUN apt-get update -qq && \
+    apt-get install -y build-essential libcurl4-openssl-dev curl && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV RAILS_ENV="production" \
+    BUNDLE_DEPLOYMENT="1" \
+    BUNDLE_PATH="/usr/local/bundle"
+
+FROM base as build
+
+RUN gem install bundler -v 2.4.7
+
+COPY Gemfile Gemfile.lock ./
+
+RUN bundle config set --local without 'development test' && bundle install
```

Comment:
Can you run this clean up after bundle install as well? from the rails dockerfile
```
RUN bundle install && \
    rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
```

https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L47